clash-ghc/src-ghc/Clash/GHC/GHC2Core.hs

```diff
@@ -730,11 +732,16 @@ coreToDataCon dc = do
              , C.dcTag         = dataConTag dc
              , C.dcType        = dcTy
              , C.dcArgTys      = repTys
+             , C.dcArgStrict   = repBangs
              , C.dcUnivTyVars  = uTvs
              , C.dcExtTyVars   = eTvs
              , C.dcFieldLabels = fLabels
              }
 
+hsImplBangToBool :: HsImplBang -> C.DcStrictness
+hsImplBangToBool HsLazy = C.Lazy
+hsImplBangToBool _ = C.Strict
```

Comment:
What about `HsUnpack`? Its documentation says it's strict too (due it's "unpackedness", I guess)

Reply:
Unpacking requires strictness, and I couldn't think of any benefit to Clash knowing if something is unpacked or not

Reply:
If you don't mind, I'd prefer this to be a complete pattern match so we get warnings if there are ever any added constructors. Other than that LGTM!